api/client_test.go

```diff
@@ -113,3 +116,70 @@ func TestClientURL(t *testing.T) {
 		}
 	}
 }
+
+func TestDoGetFallback(t *testing.T) {
+	v := url.Values{"a": []string{"1", "2"}}
+
+	type testResponse struct {
+		Values string
+		Method string
+	}
+
+	acceptPost := true
+	// Start a local HTTP server
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		req.ParseForm()
```

Comment:
error not handled here and few other places.

Maybe use  `https://github.com/prometheus/tsdb/testutil` or `https://github.com/prometheus/prometheus/testutil` to facilitate error handling in tests

`testutil.OK(t,req.ParseForm())`

Reply:
This particular one is okay, since if this fails the result won't have the form values-- so there's no need to specifically check this error.